mlir-tensorrt/compiler/include/mlir-tensorrt/Dialect/TensorRTRuntime/IR/TensorRTRuntimeOps.td

```diff
@@ -105,9 +105,54 @@ def TensorRTRuntime_EnqueueOp : TensorRTRuntime_Op<"enqueue", [
     // Declare the outs as inits/outs to DestinationStyleOpInterface.
     MutableOperandRange getDpsInitsMutable() { return getOutsMutable(); }
 
-    /// Return true if the operand at the specified index is a host tensor
-    /// argument.
-    bool isOperandOnHost(int64_t operandIdx) {
+    /// Return true if the operand is a host tensor argument.
+    bool isOperandOnHost(OpOperand *operand) {
+      unsigned operandIdx = operand->getOperandNumber();
+      if(std::optional<ArrayRef<int64_t>> indices = getHostTensorArgs()) {
+        return llvm::is_contained(*indices, operandIdx - 2);
+      }
+      return false;
+    }
+  }];
+}
+
+//===----------------------------------------------------------------------===//
+// EnqueueAllocOp
+//===----------------------------------------------------------------------===//
+
+def TensorRTRuntime_EnqueueAllocOp : TensorRTRuntime_Op<"enqueue_alloc", [
+    DeclareOpInterfaceMethods<TensorKindOpInterface>,
```

Comment:
Adding new ops requires new tests in `test/Dialect/TensorRTRuntime/invalid.mlir` and `test/Dialect/TensorRTRuntime/roundtrip.mlir`